easysocial/easysocial/friends.php

```diff
@@ -69,6 +69,27 @@ function getFriends()
 							$flag=0;						
 							$mssg=JText::_( 'PLG_API_EASYSOCIAL_NOT_SENT_REQUEST' );	
 			break;
+			case 'blocked': //getting block friends	added by pratiksha 							
+							$blk_list = $frnd_mod->getFriends($userid);
+							$flag=1;
```

Comment:
Can this logic be simplified ? Have you seen how Easysocial itself does it in their view ?